client/src/main/java/io/pravega/client/ClientConfig.java

```diff
@@ -108,6 +109,13 @@
      */
     private final MetricListener metricListener;
 
+    /**
+     * If the flag is set, a server request will time out if it doesn't receive response from Segment store
+     * in a given time period. Notice that we don't want this feature enabled in testing since it might mask some
+     * bugs.
+     */
+    private final Duration serverRequestTimeout;
```

Comment:
From this comment, not setting it disables the timeout, but we are giving it a default value right below. Could you clarify how we are disabling the timeout for tests?

Reply:
Thanks for pointing that out! Sorry it is not accurate, I have rephrased the comment. It was meant for boolean type but later it was changed to a value, so right now we just pass it a big value(3600s in system test) to disable the timeout.

Reply:
I see that the comment has changed, but I still don't understand what kind of behavior this is trying to implement. My interpretation is that:

- Optional means that the application does not need to change the default value.
- The default of 30s is a compromise between a good value for the application and a value that would make our tests fail.

Choosing an ideal value is hard, in particular, if we take into account system tests.

Reply:
Oh I see what you mean. yes we don't need to change the default value for application, however, 30s is not a compromise and it is only meant to work for application not for tests.

In system test(and some unit tests) we need to change that value and I have set it to be 3600s. See [here](https://github.com/pravega/pravega/pull/3947/files#diff-1a32a0fd61346692a52aacb1db352f8cR162). If we need to set it in other test such as longevity test, then the longevity code needs to be modifed.

Reply:
Ok, so elaborate on what you are trying to achieve then. The ideal behavior is:

- The application doesn't need to set anything and out of the box it has some reasonable timeout. How have you come up with 30s? Think that a pravega operation in general takes a few milliseconds, so 30s waiting is a long time.
- Ideally, this timeout does not mask bugs in tests, so the timeout should be for practical purposes disabled.
- Ideally, we don't have to remember to set this for new tests we build, otherwise I'm sure we will forget.

I'm not sure whether this is possible, but we could think of some sort of test flag that disables the timeout for testing and disables it for all cases in a run when the flag is set (timeout disabled). it is easier to set a flag once.

Reply:
1. yes, 30s is kind of long in terms of an operation. But I'm thinking that this timeout involves network latency and server latency, there might be a temporary network partition or the server pod has restarted and stuff like that which may takes several seconds. Also here https://github.com/pravega/pravega/blob/7685d8d2bf73b7265d09e2788d3b9751c6115c4c/client/src/main/java/io/pravega/client/segment/impl/EventSegmentReaderImpl.java#L39 is also using 30s. 

2. Yes in test mode this timeout is disabled. I have already set that in system tests.

3. For newly added system test, no we don't need to set that manually again because I have set it in the test framework that works for all system tests. 

Not exactly sure but I kind of think that the current flag is what you mean by the test flag since we only need to set once in the `clientconfig`.